pkg/clustermodule/service_test.go

```diff
@@ -81,6 +84,148 @@ func TestService_Create(t *testing.T) {
 			g.Expect(moduleUUID).To(gomega.BeEmpty())
 		})
 	})
+
+	t.Run("creation is executed", func(t *testing.T) {
+		g := gomega.NewWithT(t)
+		simr, err := initVCSimulator()
+		defer simr.Destroy()
+		g.Expect(err).ToNot(gomega.HaveOccurred())
+		md := machineDeployment("md", fake.Namespace, fake.Clusterv1a2Name)
+		md.Spec.Template.Spec.InfrastructureRef = corev1.ObjectReference{
+			Kind:      "VSphereMachineTemplate",
+			Namespace: fake.Namespace,
+			Name:      "blah-template",
+		}
+
+		machineTemplate := &infrav1.VSphereMachineTemplate{
+			TypeMeta: metav1.TypeMeta{Kind: "VSphereMachineTemplate"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "blah-template",
+				Namespace: fake.Namespace,
+			},
+			Spec: infrav1.VSphereMachineTemplateSpec{
+				Template: infrav1.VSphereMachineTemplateResource{Spec: infrav1.VSphereMachineSpec{
+					VirtualMachineCloneSpec: infrav1.VirtualMachineCloneSpec{
+						Server:       simr.ServerURL().Host,
+						Datacenter:   "*",
+						ResourcePool: "/DC0/host/DC0_C0/Resources",
+					},
+				}},
+			},
+		}
+
+		controllerCtx := fake.NewControllerContext(fake.NewControllerManagerContext(md, machineTemplate))
+		clusterCtx := fake.NewClusterContext(controllerCtx)
+		clusterCtx.VSphereCluster.Spec.Server = simr.ServerURL().Host
+		controllerCtx.ControllerManagerContext.Username = simr.Username()
+		controllerCtx.ControllerManagerContext.Password = simr.Password()
+
+		svc := NewService(controllerCtx.ControllerManagerContext, controllerCtx.Client)
+		moduleUUID, err := svc.Create(context.Background(), clusterCtx, mdWrapper{md})
+		g.Expect(err).ToNot(gomega.HaveOccurred())
+		g.Expect(moduleUUID).NotTo(gomega.BeEmpty())
```

Comment:
I think it's not worth having a separated `DoesExist` and `Delete` test functions.

These two also test teh `Create` part already. Let's combine them to a single test.


```suggestion
		g.Expect(moduleUUID).NotTo(gomega.BeEmpty())


		exists, err := svc.DoesExist(context.Background(), clusterCtx, mdWrapper{md}, moduleUUID)
		g.Expect(exists).To(gomega.BeTrue())
		g.Expect(err).NotTo(gomega.HaveOccurred())

		err = svc.Remove(context.Background(), clusterCtx, moduleUUID)
		g.Expect(err).ToNot(gomega.HaveOccurred())
```